workflow/src/main/kotlin/com/r3/corda/sdk/token/workflow/flows/UpdateDistributionList.kt

```diff
@@ -0,0 +1,93 @@
+package com.r3.corda.sdk.token.workflow.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.sdk.token.contracts.states.EvolvableTokenType
+import com.r3.corda.sdk.token.contracts.types.TokenPointer
+import com.r3.corda.sdk.token.workflow.schemas.DistributionRecord
+import com.r3.corda.sdk.token.workflow.utilities.addPartyToDistributionList
+import com.r3.corda.sdk.token.workflow.utilities.getDistributionList
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.crypto.SignedData
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.identity.Party
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.serialization.serialize
+import net.corda.core.utilities.unwrap
+import java.util.*
+import javax.persistence.criteria.CriteriaQuery
+
+object UpdateDistributionList {
+
+    @CordaSerializable
+    data class DistributionListUpdate(val oldParty: Party, val newParty: Party, val linearId: UniqueIdentifier)
+
+    // TODO It's ineffective if we are heavily using confidential identities, because we contact maintainers every time we do move
+    // tokens with token pointers.
+    @InitiatingFlow
+    class Initiator<T : EvolvableTokenType>(
+            val tokenPointer: TokenPointer<T>,
+            val oldParty: Party,
+            val newParty: Party
+    ) : FlowLogic<Unit>() {
+        @Suspendable
+        override fun call() {
+            val evolvableToken = tokenPointer.pointer.resolve(serviceHub).state.data
+            val distributionListUpdate = DistributionListUpdate(oldParty, newParty, evolvableToken.linearId)
+            val maintainers = evolvableToken.maintainers
+            val maintainersSessions = maintainers.map { initiateFlow(it) }
+            // Collect signatures from old and new parties
+            val updateBytes = distributionListUpdate.serialize()
+            val ourSig = serviceHub.keyManagementService.sign(updateBytes.bytes, oldParty.owningKey)
```

Comment:
do we need the signing part? Seems like a bit overkill :) What attack vector is the singing preventing?

Reply:
*signing

Reply:
You are right, initially I was wondering if we should remove old owners (that's why signed data structure), but it doesn't make much sense if it's fungible token.